File: video-streaming-frontend/src/app/interfaces/user.dto.ts

export interface UserDto {
  id: number;
  firstName: string;
  lastName: string;
  email?: string;
  pictureUrl?: string;
  about?: string;
  subscribersCount?: number;
  sub?: string;
}

export interface UserUpdateDto {
  firstName: string;
  lastName: string;
  about: string;
}
export interface AuthUser {
  firstName: string;
  lastName: string;
  email: string;
  sub?: string;
}

File: video-streaming-backend/src/main/java/com/fourbit/sachintha/video_streaming_platform/dto/UserDto.java
package com.fourbit.sachintha.video_streaming_platform.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDto {
	private Long id;
	private String firstName;
	private String lastName;
	private String email;
	private String pictureUrl;
	private String about;
	private String sub;
	private Boolean isRecordHistory;
}
